enhancing/enhancer.js

```diff
@@ -2,21 +2,71 @@ module.exports = {
   succeed,
   fail,
   repair,
-  get,
+  get
 };
-
+//When enhancement succeeds
+// The item's enhancement increases by 1.
+// If the item enhancement level is 20, the enhancement level is not changed.
+// The durability of the item is not changed.
 function succeed(item) {
+  
+  if(item.enhancement <= 20){
+    item.enhancement +=1;
+    
+  }
   return { ...item };
 }
 
+//If the item's enhancement is less than 15, the durability of the item is
+//decreased by 5.
+//If the item's enhancement is 15 or more, the durability of the item is
+// decreased by 10.
+//If the item's enhancement level is greater than 16, the enhancement level
+//decreases by 1 (17 goes down to 16, 18 goes down to 17).
 function fail(item) {
+  if (item.enhancement < 15) {
+    item.durability -= 5;
+  } else if (item.enhancement >= 15 && item.enhancement <= 16) {
+    item.durability -= 10;
+  } else if (item.enhancement <= 20) {
+    item.enhancement -= 1;
+  }
   return { ...item };
 }
 
 function repair(item) {
+  item.durability = 100;
   return { ...item };
 }
 
 function get(item) {
-  return { ...item };
+   if(item.enhancement>0){
+      item.name="[+" + item.enhancement +"]"+item.name
+      return { ...item };
+   }
+   return item;
```

Comment:
Why return the same object here but return a new object up above? It seems like you might end up with some reference problems somewhere depending on how these are used. [Here is a good article for that](https://codeburst.io/explaining-value-vs-reference-in-javascript-647a975e12a0).